src/types/state.rs

```diff
@@ -1,12 +1,17 @@
 use std::fmt;
 use std::io;
 use std::str;
+use std::str::FromStr;
+
+use serde::{Deserialize, Deserializer, Serialize};
+use serde::de::{Error, Unexpected};
 
 /// Possible battery state values.
 ///
 /// Unknown can mean either controller returned unknown,
 /// or not able to retrieve state due to some error.
-#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+#[derive(Debug, Eq, PartialEq, Copy, Clone, Serialize, schemars::JsonSchema)]
```

Comment:
If this is optional, you need to use `cfg_attr` like the configs in starship do.